### PR TITLE
Add support for cachedMem and fix usedMem on OpenBSD.

### PR DESCRIPTION
**Pull request by juanfra684**
Tested against htop 2.0 on OpenBSD 5.9 -current. In my tests, the numbers in the meters are quite similar to the output of top (unlike htop, top shows the memory occupied by the cache as used).

The one-line change in OpenBSDProcessList.h is for to avoid to run MakeHeader.py during the build phase.

/cc @mmcco

### CODE REVIEW

**Comment by mmcco** [comment]
Thanks for this - it looks good. I'll probably have time to test tomorrow. In the meantime, a few quick comments:

* Be sure to adhere to the existing style. It isn't KNF, but... it's something. That includes three-space indents (see line 116).
* It probably isn't necessary to add yourself to the header copyright.   :-)
* I think we should keep the includes alphabetized. That said, they probably need to be rearranged in all files in a separate PR.

**Comment by juanfra684** [comment]
I updated the commit with your feedback. Thanks! :)

It took me hours to get the correct value for `usedMem`. I'm still not 100% convinced.

**Comment by mmcco** [comment]
> It took me hours to get the correct value for usedMem. I'm still not 100% convinced.

Welcome to system monitoring code.  :-)  I try to pull functions from top(1) and ps(1) whenever possible.

File: openbsd/OpenBSDProcessList.c [hunk 1]
```diff
@@ -95,16 +96,29 @@ void ProcessList_delete(ProcessList* this) {
...
+   size_t size_uvmexp = sizeof(uvmexp);
 
-   if (sysctl(uvmexp_mib, 2, &uvmexp, &size, NULL, 0) < 0) {
+   if (sysctl(uvmexp_mib, 2, &uvmexp, &size_uvmexp, NULL, 0) < 0) {
       err(1, "uvmexp sysctl call failed");
    }
 
    //kb_pagesize = uvmexp.pagesize / 1024;
-   pl->usedMem = uvmexp.active * pageSizeKb;
+   //pl->usedMem = uvmexp.active * pageSizeKb;
```
Comment (hishamhm) [inline openbsd/OpenBSDProcessList.c]: No need to keep old code around as comments; we can always refer to version control to look at the old version. :)

**Comment by hishamhm** [comment]
Thanks for the review @mmcco! Once you give it a spin, let me know and I'll merge.

**Comment by falsovsky** [comment]
I am currently using this, and it seems good.

**Comment by hishamhm** [comment]
Merged, thank you!

**Comment by mmcco** [comment]
Thanks for merging - I forgot to come back to this. I'll test it out now.

**Pull request merged**